### ISSUE TITLE
[ITensors] [BUG] Commit 56e8a1ed7cf31c08c0d9e26d40ab5d34197af56d breaks autompo unit tests

### ISSUE THREAD

**Issue by JanReimers** [ITensor/ITensors.jl#1006]
**Description of bug**
ITensors.jl version 0.3.13

This commit from [WIP] [ITensors] [ENHANCEMENT] Remove `OpSum` in favor of `Ops.OpSum` (#920)
results in a failure in the test/autompo.jl unit tests.

**Minimal code demonstrating the bug or unexpected behavior**
julia test/autompo.jl

Stack dump:
Matrix operator representation - hashing bug: Error During Test at /home/janr/.julia/dev/ITensors/test/autompo.jl:1091
  Got exception outside of a @test
  ArgumentError: No file exists at given path: opsum_hash_bug.jld2
  Stacktrace:
    [1] checkpath_load
      @ ~/.julia/packages/FileIO/aP78L/src/loadsave.jl:167 [inlined]
    [2] load(::String, ::String, ::Vararg{String}; options::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
      @ FileIO ~/.julia/packages/FileIO/aP78L/src/loadsave.jl:110
    [3] load(::String, ::String, ::Vararg{String})
      @ FileIO ~/.julia/packages/FileIO/aP78L/src/loadsave.jl:110
    [4] macro expansion
      @ ~/.julia/dev/ITensors/test/autompo.jl:1092 [inlined]

@testset "Matrix operator representation - hashing bug" begin
    comps, n, dims = load("opsum_hash_bug.jld2", "comps", "n", "dims")  #line 1092
    s = [Index(d) for d in dims]
    for _ in 1:100
      os = components_to_opsum(comps, n)
      # Before defining `hash(::Op, h::UInt)`, this
      # would randomly throw an error due to
      # some hashing issue in `MPO(::OpSum, ...)`
      MPO(os, s)
    end
  end

code tries to load a pr-existing file that does not exist.  The file opsum_hash_bug.jld2 either needs to be created inside the test framework (preferable IMHO), or included in the source tree (not preferable IMHO).

**Comment by mtfishman** [comment]
That file is stored in the repository: https://github.com/ITensor/ITensors.jl/blob/main/test/opsum_hash_bug.jld2

It may be that when you run the test in that way it doesn't find the file because the `load` call here:
https://github.com/ITensor/ITensors.jl/blob/214915c7d5b1385101dcb663dd43d3591bfc4ee4/test/autompo.jl#L1097
is using a relative path instead of an absolute path, we could make that an absolute path to make the test more robust.

**Comment by JanReimers** [comment]
Yeah OK, I see it.  As far as I can tell this is the only unit test that requires pwd==.../ITensors/test.  I think a relative path is better than an absolute path.  I only saw the problem because my IDE (vscode) defaults to running .jl files with pwd=.../ITensors regardless of where .jl is situated in tree.  I guess if other developers run into this it might be worth finding the relative path to opsum_hash_bug.jld2 before the load call to make things more robust.  Closing for now.
